docs/zedctr.md

```diff
@@ -0,0 +1,96 @@
+* It has `net: host`, working directly in the host networking namespace.
+* It mounts most of the root filesystem into its own namespace, including `/:/hostfs`. The mounts are listed below:
+
+```yml
+  binds:
+    - /lib/modules:/lib/modules
+    - /dev:/dev
+    - /etc/resolv.conf:/etc/resolv.conf
+    - /run:/run
+    - /var/config:/config
+    - /var/log:/var/log/dom0
+    - /:/hostfs
+```
+
+## Components
+
+The image `zededa/zedctr` is built from its [Dockerfile](../pkg/zedctr/Dockerfile), which is generated from its [Dockerfile.in](../pkg/zedctr/Dockerfile.in), with the relevant image names and tags patched in.
+
+It is built from the following packages, installed into the following locations:
+
+|Package|Image|Source|Install|Purpose|Comments|
+|---|---|---|---|---|
+|ztools|`zededa/ztools`|[go-provision](https://github.com/zededa/go-provision)|`/`|Interface with zedcloud||
+|lisp|`zededa/lisp`|[lisp](https://github.com/zededa/lisp)|[Locator/Identifier Separation Protocol](https://en.wikipedia.org/wiki/Locator/Identifier_Separation_Protocol)|see list [here](#lisp-install)|implementation for mesh networking|custom implementation in [go](https://golang.org), as the reference implementation is in python and has unneeded services|
+|xen tools|`zededa/xen-tools`|[pkg/xen-tools](../pkg/xen-tools)|`/`|Xen utilities||
+|dnsmasg|`zededa/dnsmasq`|[pkg/dnsmasq](../pkg/dnsmasq)|`/usr/sbin/dnsmasq` to `/opt/zededa/bin/dnsmasq`|local DNS server|Unclear why we need a local build|
+|strongswan|`zededa/strongswan`|[pkg/strongswan](../pkg/strongswan)|`/`|IPSec VPN|Dockerfile has [comment](../pkg/strongswan/Dockerfile#L63-L66), stating that we build custom because of libressl conflict with curl-dev. Should be investigated if this remains an issue|
+|gpt tools|`zededa/gpt-tools`|[pkg/gpt-tools](../pkg/gpt-tools)|`/`|Utilities for working with GPT partitioned filesystems. See [here](./UPSTREAMING.md#gpt-tools) for more details on what is in this package|
+|watchdog|`zededa/watchdog`|[pkg/watchdog](../pkg/watchdog)|`/usr/sbin` to `/usr/sbin`|[watchdog]() daemon to watch for userspace processes and, if they are not providing proper heartbeats, restart the system|Unclear why we need to restart the system vs restarting zombie processes, a la systemd/runit/supervisord/etc.|
+
+In addition, a few local installations are provided as necessary, defined in the [Dockerfile](../pkg/zedctr/Dockerfile). They are:
+
+* `libfdt` via `apk` for `arm64` only
+* symlink `/usr/lib/libpcap.so` to `libpcap.so.1`
+* generate a host ssh key, configure `/etc/ssh/sshd_config` to permit root login via ssh, and change the ssh root key. It is unclear why we are doing this for several reasons:
+    * We are generating the root password at package build time, storing secrets in a static OCI image. 
+    * We are generating the ssh host key at package build time, causing all hosts to have the same ssh host key.
+    * We are allowing root login via ssh
```

Comment:
The root ssh login are disabled by default.
Can be enabled for debugging using the device API. See 
https://github.com/zededa/go-provision/blob/fixes/docs/global-config-variables.md

Reply:
But even for debug we should have something more secure, like a trusted ssh public key (and the private key is not in the built image).

Reply:
Let me see if I get this. The goal is to be able to ssh into a node, if needed, for real-time debug purposes. This would be enabled/disabled via the api (per the reference link you sent).

Is that correct?

If I did understand the purpose correctly, then I still don't understand the rest of it:

> The root ssh login are disabled by default.

But we permitted it in `/etc/ssh/sshd_config`. Do you mean that it is disabled on the per-user basis?

> the private key is not in the built image

You mean not, "the private key is not built in now," as it is built in now. You mean, "if we have a trusted ssh public key, then the private key would not be built into the image". 

Is that right?

Reply:
Updated. Please take a look.